packages/offix-client/src/OfflineClient.ts

```diff
@@ -141,22 +147,22 @@ export class OfflineClient {
     // Optimistic Responses
     // TODO move this somewhere that makes sense, also handle error cases
     this.queue.registerOfflineQueueListener({
-      onOperationEnqueued: (operation: QueueEntryOperation<MutationOptions>) => {
+      onOperationEnqueued: (operation: ApolloQueueEntryOperation) => {
```

Comment:
ok ok.. but we wanted to get generic later on. 
This will straightaway mean that we going to break stuff. 
Any reason why this needs to be apollo?

Reply:
I don't really see what's the problem here. OfflineClient is a specific implementation of the generic Queue + Store mechanisms.

Reply:
Later down the line we will have to untangle some small amount of logic from the `OfflineClient` and that logic + generic queue + generic store will become a scheduler package.

At that point OfflineClient will simply become a concrete implementation of the scheduler using all ApolloSpecific types. Does that make sense?

Reply:
The Queue and Store itself are still completely generic though. IMO there is no need for a user of `offix-client` to care about those being generic because they are using the very Apollo specific bundle.

The main benefit of the generics is for maintenance and for advanced cases like implementation of new schedulers (e.g. rest/custom)

Reply:
Ok. I missed that we want OfflineClient to be Apollo specific. It makes a lots of sense from many aspects including compatibility and support.